pkg/azureclients/diskclient/azure_diskclient.go

```diff
@@ -470,3 +476,11 @@ func (page DiskListPage) Values() []compute.Disk {
 	}
 	return *page.dl.Value
 }
+
+func recreateClientChecker(response *http.Response) {
```

Comment:
Can we extract this into util.go?

Reply:
I think to put in this in util.go I'd need to call [InitializeCloudFromConfig](https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure.go#L489) because I wouldn't know the client we're using at the time of the calls and just reset them all. I'm not confident on this approach or impact though. Or possibly [NewCloud](https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/provider/azure.go#L377) is better to call?